rcl/include/rcl/publisher.h

```diff
@@ -336,6 +340,29 @@ RCL_WARN_UNUSED
 rmw_publisher_t *
 rcl_publisher_get_rmw_handle(const rcl_publisher_t * publisher);
 
+/// Check that the publisher is valid
+/**
+ * The bool returned is `false` if:
+ *   - the argument `publisher` is `NULL`
+ *   - the argument's `publisher->impl` is `NULL`
```

Comment:
Is this sufficient?

```c
/**
 * The bool returned is `false` if the argument `publisher` is invalid.
 * The bool returned is `true` otherwise.
 */
```

(not leaving out the details beneath, but I didn't retype them here for clarity)